cpp/ql/lib/experimental/quantum/OpenSSL/Operations/CipherOperation.qll

```diff
@@ -0,0 +1,271 @@
+import experimental.quantum.Language
+private import OpenSSLOperationBase
+private import experimental.quantum.OpenSSL.AlgorithmValueConsumers.OpenSSLAlgorithmValueConsumers
+import EVPPKeyCtxInitializer
+
+/**
+ * A base class for all EVP cipher operations.
+ */
+abstract class EvpCipherInitializer extends OperationStep {
+  override DataFlow::Node getInput(IOType type) {
+    result.asExpr() = this.getArgument(0) and type = ContextIO()
+    or
+    result.asExpr() = this.getArgument(1) and
+    type = PrimaryAlgorithmIO() and
+    // Null for the algorithm indicates the algorithm is not actually set
+    // This pattern can occur during a multi-step initialization
+    // TODO/Note: not flowing 0 to the sink, assuming a direct use of NULL for now
```

Comment:
Suggest refactoring the comment for clarity:
// Constants that are not equal to zero OR 
// Non-constants (e.g., variable accesses, which require data-flow to determine the value)